api/src/main/java/io/strimzi/api/kafka/model/listener/IngressListenerBrokerConfiguration.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2018, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model.listener;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import io.strimzi.crdgenerator.annotations.Description;
+import io.sundr.builder.annotations.Buildable;
+import lombok.EqualsAndHashCode;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Configures broker ingress
+ */
+@JsonPropertyOrder({"broker", "advertisedHost", "advertisedPort", "host"})
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@Buildable(
+    editableEnabled = false,
+    generateBuilderPackage = false,
+    builderPackage = "io.fabric8.kubernetes.api.builder"
+)
+@EqualsAndHashCode(callSuper = true)
+public class IngressListenerBrokerConfiguration extends ExternalListenerBrokerOverride {
```

Comment:
Since this has exactly the same properties as `IngressListenerBootstrapConfiguration` couldn't they be the same class? Or an abstract baseclass and two concrete subclasses if there's a benefit from the extra type safety?

Reply:
This inherits from the `ExternalListenerBrokerOverride` class which unlike the bootstrap has some more options. In particular the broker` fields with the broker ID. So I do not think we can create any base class here.

Reply:
Ah, I missed that. Fair enough.